### PR TITLE
Simplify backend CI

### PR DESCRIPTION
**Pull request by rracanicci**
- Run poetry check, lint and test as a step in the backend dockerfile
- Simplify python CI

### CODE REVIEW

File: python/backend/Dockerfile [hunk 1]
```diff
@@ -44,5 +44,10 @@ RUN poetry config virtualenvs.create false \
 # Copy remaining files
 COPY . .
 
+# Check formatting, lint and tests
+RUN poetry run poe format_check \
```
Comment (darthtrevino) [inline python/backend/Dockerfile:48]: can we add a custom poe task `verify` that will run these steps sequentially?

File: .github/workflows/ci-python.yml [hunk 2]
```diff
@@ -11,49 +11,10 @@ on:
...
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python-version: [3.8]
-        poetry-version: [1.2.1]
 
     steps:
       - uses: actions/checkout@v2
 
-      - name: Set up Python ${{ matrix.python-version }}
```
Comment (darthtrevino) [inline .github/workflows/ci-python.yml:22]: Interesting approach, let's try it out

**Review by darthtrevino — APPROVED** [review]
Interesting idea

File: python/backend/Dockerfile (hunk 1 above)
Comment (rracanicci) [reply python/backend/Dockerfile:48]: sure thing, just updated the PR with it :)

**Pull request merged**